simpref/simpref.go

```diff
@@ -0,0 +1,158 @@
+package simpref
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/bitrise-io/go-plist"
+	"github.com/bitrise-io/go-utils/v2/fileutil"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-utils/v2/pathutil"
+	"github.com/bitrise-io/go-xcode/v2/destination"
+)
+
+const (
+	DefaultIPhoneSimulatorPreferencesPth = "~/Library/Preferences/com.apple.iphonesimulator.plist"
+	defaultSimulatorDestination          = "platform=iOS Simulator,name=Bitrise iOS default,OS=latest"
+)
+
+type IPhoneSimulatorPreferences struct {
+	pth         string
+	format      int
+	preferences map[string]any
+
+	deviceFinder destination.DeviceFinder
+	fileManager  fileutil.FileManager
+	pathModifier pathutil.PathModifier
+	logger       log.Logger
+}
+
+func OpenIPhoneSimulatorPreferences(pth string, deviceFinder destination.DeviceFinder, pathModifier pathutil.PathModifier, fileManager fileutil.FileManager, logger log.Logger) (*IPhoneSimulatorPreferences, error) {
+	absPth, err := pathModifier.AbsPath(pth)
+	if err != nil {
+		return nil, err
+	}
+
+	prefsFile, err := fileManager.Open(absPth)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, fmt.Errorf("failed to open file: %w", err)
+		}
+
+		logger.Debugf("iphonesimulator preferences file not found at %s, creating one...", absPth)
```

Comment:
I am just thinking out load here but I feel like this could be a log something else than debug. Apple sometimes silently moves stuff around and seeing this line in the build logs could make someone suspicious that something changed with the newer Xcode version.

I think a warning is too much for this innocent line. So maybe a regular `Printf`?

Reply:
Updated [here](https://github.com/bitrise-steplib/steps-disable-iphonesimulator-connect-hardware-keyboard/pull/1/commits/9c5fc5f8db0a106d98c5a38ec1baec60a3d1f7b2), and also updated the logs of the DeviceFinder dependency, because it printed a quite verbose log with seemingly useless warnings.

Reply:
Great idea.